packages/devtools_app/lib/src/shared/banner_messages.dart

```diff
@@ -416,6 +417,60 @@ For the most accurate absolute memory stats, relaunch your application with the
   }
 }
 
+class UnsupportedFlutterVersionWarning {
+  const UnsupportedFlutterVersionWarning({
+    required this.screenId,
+    required this.currentFlutterVersion,
+    required this.supportedFlutterVersion,
+  });
+
+  final String screenId;
+
+  final FlutterVersion currentFlutterVersion;
+
+  final SemanticVersion supportedFlutterVersion;
+
+  BannerMessage build(BuildContext context) {
+    return _BannerWarning(
+      key: Key('UnsupportedFlutterVersionWarning - $screenId'),
+      textSpans: [
+        TextSpan(
+          text: 'This version of DevTools expects the connected app to be run'
+              ' on Flutter >= $supportedFlutterVersion, but the connected app'
+              ' is running on Flutter $currentFlutterVersion. Some'
+              ' functionality may not work. If this causes issues, try'
+              ' upgrading your Flutter version.',
+          style: TextStyle(
+            color: _BannerWarning.foreground,
+            fontSize: defaultFontSize,
+          ),
+        ),
+      ],
+      screenId: screenId,
+    );
+  }
+}
+
+void maybePushUnsupportedFlutterVersionWarning(
+  BuildContext context,
+  String screenId, {
+  required SemanticVersion supportedFlutterVersion,
+}) {
+  if (offlineController.offlineMode.value) return;
+  if (serviceManager.connectedApp?.isFlutterAppNow ?? false) {
```

Comment:
Optional suggestion: return early if not a Flutter app to avoid the nested if-clauses

Reply:
done.